packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config.vue

```diff
@@ -411,55 +528,104 @@ const onUpdateSelection = (id) => {
 	emit('update-output-port', outputPort);
 };
 
+// Creates a temp config (if doesnt exist in state)
+// This is used for beaker context when there are no outputs in the node
+const createTempModelConfig = async () => {
+	const state = _.cloneDeep(props.node.state);
+	if (state.tempConfigId !== '' || !model.value) return;
+	const data = await createModelConfiguration(
+		model.value.id,
+		'Temp_config_name',
+		'Utilized in model config node for beaker purposes',
+		model.value,
+		true
+	);
+
+	state.initials = knobs.value.initials;
+	state.parameters = knobs.value.parameters;
+	state.timeseries = knobs.value.timeseries;
+	state.tempConfigId = data.id;
+	emit('update-state', state);
+};
+
+// Fill the form with the config data
+const initialize = async () => {
+	const state = _.cloneDeep(props.node.state);
```

Comment:
I feel like we should clean up the initialize section a bit, first you don't seem to be updating state here, so no need to clone?

But further more, it is a bit confusing here, because if I were to unroll `createTempModelConfig`, you have
- state updating knob.values, and
- knob.values updating state
While this may end up being the same thing mathematically (I don't actually know) it would be easier to reason with if we have a more uni-directional flow.

It is also weird that in `createTempModelConfig` you have state updated via knobs, as opposed to from the model-configuration you just created.

Let me know if you'd like to have a chat re: organizing.